### PR TITLE
216 Added contributing.md

### LINKED ISSUE #216: Add CONTRIBUTING.md
The file should explain how to create issues, branches; which coding standards we're using; how a good commit message should look like; etc.

Examples:
- https://github.com/wp-cli/wp-cli/blob/master/CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by javorszky**
Closes #216

### CODE REVIEW

**Comment by javorszky** [comment]
Added https://github.com/GlotPress/GlotPress-WP/commit/3d0cfc3395f1a65dbd4c26e34930340e4d8fbad0 that fixes the issues flagged up (except the composer one).

**Comment by ocean90** [comment]
@toolstack If it looks good to you feel free to merge it.

If possible squash the commits before the merge.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# Contribute
+
+## Setting up
+
+1. Clone this git repository on your local machine inside a `wp-content/plugins/` directory
```
Comment (toolstack) [inline CONTRIBUTING.md]: Perhaps:  "1. Clone this git repository to your development system, this is commonly inside your WordPress development site in the 'wp-content/plugins' directory."

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+2. Install [PHPUnit](https://phpunit.de/). We recommend using version 4.5, as that's the most current version that is compatible with all PHP versions we check against, and thus Travis is using that version.
+3. Activate the plugin.
+4. Visit `http://local.sitename/glotpress` to start using GlotPress. `local.sitename` is the domain where your local development WordPress site runs, dependent on configuration.
```
Comment (toolstack) [inline CONTRIBUTING.md]: Replace "local" with "development".

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+## Submitting patches
+
+Whether you want to fix a bug or implement a new feature, the process is pretty much the same:
+
+0. [Search existing issues](https://github.com/GlotPress/GlotPress-WP/issues); if you can't find anything related to what you want to work on, [open a new issue](#helpful-tips-for-writing-issues).
+1. [Fork](https://github.com/GlotPress/GlotPress-WP/fork) the repository.
+2. Create a branch for each issue you'd like to address. Base your new branch on top of `develop` branch. Branches are named as such: `ISSUEID-keywords` for branches where an issue exists, or `keywords` for a branch without an issue yet. For example issue 216 for adding contributing.md would be `216-add-contributing-md`. The same branch would be named `add-contributing-md` if there wasn't an issue for it. Commit your changes.
```
Comment (toolstack) [inline CONTRIBUTING.md]: Should probably just remove: 

"The same branch would be named `add-contributing-md` if there wasn't an issue for it."

They should have opened a ticket for it back on step 0 :)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+3. Push the code changes from your local clone to your fork.
+4. Open a pull request from your fork's feature branch to GlotPress's `develop` branch. Pull request name should be `ISSUEID-keywords`. Take a look at [one such pull request](https://github.com/GlotPress/GlotPress-WP/pull/241) to get an idea of it.
```
Comment (toolstack) [inline CONTRIBUTING.md]: Add a note to include the "Resolves #" comment in the PR or move the stuff below to here so it's top of mind.

Maybe change "Pull request name should be `ISSUEID-keywords`." to " Pull request name match the branch name."?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
+# Contribute
+
```
Comment (toolstack) [inline CONTRIBUTING.md:2]: Need to add a bit of introduction, something like:

There are several ways you can contribute to GlotPress and all are appreciated:
1. Submitting patches: whether fixing a bug or adding new functionality
2. Testing patches / PR's
3. Running and writing tests

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+We use the [Git workflow](https://github.com/GlotPress/GlotPress-WP/wiki/5.-Git-workflow). Please have a read through, as it will make everyone's life easier. Should you have questions about that, reach out to us on Slack (see below).
+
+Pull request should have `Fixes #issue` / `Closes #issue`, or, if it's a partial solution, `Part of #issue` in the message to [enable automatic issue closing when a PR is merged](https://help.github.com/articles/closing-issues-via-commit-messages/).
+
+It doesn't matter if the code isn't perfect. The idea is to get it reviewed early and iterate on it.
+
+Lastly, please follow the [WordPress Coding Standards](http://make.wordpress.org/core/handbook/coding-standards/).
+
+## Testing patches / PRs
+
```
Comment (toolstack) [inline CONTRIBUTING.md:33]: Add:

Testing of patches and PRs is a critical part of development, if you have contributed a PR or just want to help out testing is a great place to start.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+1. Check out the branch the PR was submitted on.
+2. Leave feedback on the PR. Write up what you tested, what your results were and whether it solves the issue or not.
+
+If you need help with checking out the branch, or the PR was submitted from someone's fork, GitHub provides help in checking out the correct branch. Navigate to the PR in question, and you should see a link to command line instructions. Follow those instructions.
+
+![](https://cloud.githubusercontent.com/assets/617637/12953405/9df99808-d01a-11e5-88eb-958d23871f87.png)
+
+## Running and writing tests
+
```
Comment (toolstack) [inline CONTRIBUTING.md:44]: Add: 
GlotPress includes automated test code but coverage is not complete at this time.  Contributing new tests for GlotPress is a great way to better understand the codebase and contribute to the project at the same time.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+There are two types of automated tests:
+
+* unit tests, implemented using [PHPUnit](http://phpunit.de/)
+* general code metrics using [Scrutinizer](https://scrutinizer-ci.com/)
+
+### Unit tests
+
+The unit test files are in the `tests/` directory.
+
```
Comment (toolstack) [inline CONTRIBUTING.md:55]: Add:

You can run PHPUnit on Unix or Windows however at this time instructions and scripts are only provided for Unix.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+To run the unit tests, just execute:
+
+```bash
```
Comment (toolstack) [inline CONTRIBUTING.md]: Get rid of `bash`?

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+There are two types of automated tests:
+
+* unit tests, implemented using [PHPUnit](http://phpunit.de/)
+* general code metrics using [Scrutinizer](https://scrutinizer-ci.com/)
+
+### Unit tests
+
+The unit test files are in the `tests/` directory.
+
+To run the unit tests, just execute:
```
Comment (toolstack) [inline CONTRIBUTING.md]: "To run the unit tests, open a command shell, change to your development directory for GlotPress and run:"

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+
+To write unit tests, find the relevant file that has similar tests already in them (they are named accordingly), and add the tests there. If there isn't a file that has tests for the functionality you've written, create a new file, and name it `test_<functionality>.php`. PHPUnit will pick up the file automatically. Refer to the [PHPUnit documentation](https://phpunit.de/manual/4.5/en/index.html) and existing files for how to write new tests.
+
+### Scrutinizer
+
+For the most part feedback from Scrutinizer will flag up potential issues with the submitted code. It will tell you whether anything has been introduced or fixed. It will also give you explanations of those.
+
+## Helpful tips for writing issues
+
+In order to best help you / verify bug reports, we need a few details in the issue to move forward.
```
Comment (toolstack) [inline CONTRIBUTING.md]: Maybe: "When submitting an issue on GitHub there are several things you should include to ensure we can verify the problem and have enough information to resolve it."

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+> I expect the translatable string to be `Some translatable string with a %1$s`, but instead it's `Some translatable string with a`. Placeholder is missing.
+
+(not an actual bug)
+
+Look at some of the issues open currently to get an idea of it.
+
+### 2. It helps to know what your environment is like
+
+* Which PHP version are you using?
+* Which version of GlotPress do you have? (Which is the last commit? Find it with `git log -1`)
```
Comment (toolstack) [inline CONTRIBUTING.md:104]: Add: WP Version, WP plugins, GP plugins, MySQL version.

**Comment by javorszky** [comment]
Thank you @ocean90 and @toolstack! Fixed all the issues and squashed everything into one commit.

**Pull request merged**